src/components/Analyzer.js

```diff
@@ -1,6 +1,60 @@
-const Pitchfinder = require("pitchfinder");
-const detectPitch = Pitchfinder.AMDF();
+import React, { Component } from 'react';
+import Pitchfinder from 'pitchfinder';
+// import a component if you want to return it in your render
 
-const myAudioBuffer = getAudioBuffer(); // assume this returns a WebAudio AudioBuffer object
-const float32Array = myAudioBuffer.getChannelData(0); // get a single channel of sound
-const pitch = detectPitch(float32Array); // null if pitch cannot be identified
+class Analyzer extends Component {
+  constructor(){
+    super();
+    this.state={
+      processor: null
+    }
+
+  }
+
+  componentDidMount(){
+    this.runAnalyzer();
+  }
+
+  componentWillUnmount() {
+    if (this.state.processor){this.state.processor.disconnect()}; // without this, error: Cannot read property 'disconnect' of null
```

Comment:
Why? Is there an underlying issue that might be causing this? I noticed that when the component unmounts the processor continues to log in the console. Is there a way to stop it when the component gets unmounted?